addons/pinniped/config-controller/controllers/controllers.go

```diff
@@ -76,156 +62,150 @@ func (c *PinnipedController) SetupWithManager(manager ctrl.Manager) error {
-		if !labelExists {
-			err := errors.New("could not find ca bundle")
-			log.Error(err, "error retrieving ca bundle from pinniped-info configmap")
-			return err
-		}
-	} else {
-		supervisorAddress = ""
-		supervisorCABundle = ""
+
+	if err := c.reconcileDataValues(ctx, secret, cluster, pinnipedInfoCM); err != nil {
+		return err
 	}
 
-	log.V(1).Info("retrieved data from pinniped-info configmap", "supervisorAddress", supervisorAddress, "supervisorCABundle", supervisorCABundle)
+	return nil
+}
 
-	pinnipedAddonSecret := &corev1.Secret{
+func (c *PinnipedController) reconcileDataValues(ctx context.Context, secret *corev1.Secret, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) error {
+	log := c.Log.WithValues(constants.SecretNamespaceLogKey, secret.Namespace, constants.SecretNameLogKey, secret.Name)
+	pinnipedSecret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Namespace: cluster.Namespace,
-			Name:      fmt.Sprintf("%s-pinniped-addon", cluster.Name),
-			Labels: map[string]string{
-				constants.TKGAddonLabel:       constants.PinnipedAddonLabel,
-				constants.TKGClusterNameLabel: cluster.Name,
-			},
-			Annotations: map[string]string{
-				constants.TKGAddonTypeAnnotation: constants.PinnipedAddonTypeAnnotation,
-			},
+			Namespace: secret.Namespace,
+			Name:      secret.Name,
 		},
 	}
-	// TODO: remove this
-	pinnipedAddonSecret.Type = constants.TKGAddonType
-	pinnipedAddonSecret.Data = make(map[string][]byte)
-
-	pinnipedAddonSecretMutateFn := func() error {
-		// TODO: do we need to add these fields?!?!?!?!?!?:
-		//    pinniped:
-		//      cert_duration:
-		//      cert_renew_before:
+
+	log.V(1).Info("creating or patching secret")
+	// TODO: Create or Patch here vs. just patch since it should already be there?
+	result, err := controllerutil.CreateOrPatch(ctx, c.client, pinnipedSecret, c.getMutateFn(pinnipedSecret, cluster, pinnipedInfoCM))
+	if err != nil {
+		log.Error(err, "error creating or patching data values")
+		return err
+	}
+
+	log.Info("finished creating/patching", "result", result)
+
+	return nil
+}
+
+func (c *PinnipedController) getMutateFn(secret *corev1.Secret, cluster *clusterapiv1beta1.Cluster, pinnipedInfoCM *corev1.ConfigMap) func() error {
+	return func() error {
+		log := c.Log.WithValues(constants.SecretNamespaceLogKey, secret.Namespace, constants.SecretNameLogKey, secret.Name)
+
+		supervisorAddress := ""
+		supervisorCABundle := ""
+		identityManagementType := constants.None
+
+		if pinnipedInfoCM.Data != nil {
+			var (
+				issuerExists bool
+				bundleExists bool
+			)
+			supervisorAddress, issuerExists = pinnipedInfoCM.Data[constants.IssuerKey]
+			supervisorCABundle, bundleExists = pinnipedInfoCM.Data[constants.IssuerCABundleKey]
+			//TODO: do we only want to set idmgmttype to oidc if both issuer/cabundle exist?
+			if issuerExists && bundleExists {
+				identityManagementType = constants.OIDC
+			}
```

Comment:
Also wanted to know what y'all thought of this (leaving `identity_management_type: none` if neither the `issuer` nor the `caBundle` exist in the pinniped info configmap.

Reply:
That seems to make sense.  Also, if one exists but not the other, ought we log/error?  That seems to be a weird state.